op-node/rollup/derive/channel_bank.go

```diff
@@ -127,23 +130,40 @@ func (cb *ChannelBank) Read() (data []byte, err error) {
 		cb.metrics.RecordChannelTimedOut()
 		delete(cb.channels, first)
 		cb.channelQueue = cb.channelQueue[1:]
-		// There is a new head channel if there is a channel after we have removed the first channel
-		if len(cb.channelQueue) > 0 {
-			cb.metrics.RecordHeadChannelOpened()
-		}
 		return nil, nil // multiple different channels may all be timed out
 	}
-	if !ch.IsReady() {
-		return nil, io.EOF
+
+	// At the point we have removed all timed out channels from the front of the channelQueue.
+	// Pre-Canyon we simply check the first index.
+	// Post-Canyon we read the entire channelQueue for the first ready channel. If no channel is
+	// available, we return `nil, io.EOF`.
+	if !cb.cfg.IsCanyon(cb.Origin().Time) {
```

Comment:
Documented. It's a little tricky with the timestamps because I think anything which is from L1 (batches, frames, channel bank / batch queue) should activate on L1 timestamps and anything which is L2 (attributes, engine API calls) should activate based on the L2 timestamp.